### ISSUE TITLE
ts errors with vite5

### ISSUE THREAD

**Issue by morriq** [relative-ci/rollup-plugin-webpack-stats#156]
You can find issue here, IDE should highlight `webpackStats` after installation of packages. 

https://stackblitz.com/edit/withastro-astro-vgczpf?file=astro.config.ts

```
  Error: astro.config.ts(28,15): error TS2322: Type 'Plugin<any>' is not assignable to type 'PluginOption'.
    Type 'import("/home/runner/work/code/code/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/rollup").Plugin<any>' is not assignable to type 'import("/home/runner/work/code/code/node_modules/.pnpm/vite@5.0.7/node_modules/vite/dist/node/index").Plugin<any>' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
      Types of property 'resolveId' are incompatible.
        Type 'ObjectHook<(this: PluginContext, source: string, importer: string | undefined, options: { assertions: Record<string, string>; custom?: CustomPluginOptions; isEntry: boolean; }) => ResolveIdResult | Promise<...>, {}>' is not assignable to type 'ObjectHook<(this: PluginContext, source: string, importer: string | undefined, options: { attributes: Record<string, string>; custom?: CustomPluginOptions; ssr?: boolean; isEntry: boolean; }) => ResolveIdResult | Promise<...>>'.
          Type '(this: PluginContext, source: string, importer: string | undefined, options: { assertions: Record<string, string>; custom?: CustomPluginOptions; isEntry: boolean; }) => ResolveIdResult | Promise<...>' is not assignable to type 'ObjectHook<(this: PluginContext, source: string, importer: string | undefined, options: { attributes: Record<string, string>; custom?: CustomPluginOptions; ssr?: boolean; isEntry: boolean; }) => ResolveIdResult | Promise<...>>'.
            Type '(this: PluginContext, source: string, importer: string | undefined, options: { assertions: Record<string, string>; custom?: CustomPluginOptions; isEntry: boolean; }) => ResolveIdResult | Promise<...>' is not assignable to type '(this: PluginContext, source: string, importer: string | undefined, options: { attributes: Record<string, string>; custom?: CustomPluginOptions; ssr?: boolean; isEntry: boolean; }) => ResolveIdResult | Promise<...>'.
              The 'this' types of each signature are incompatible.
                Property 'moduleIds' is missing in type 'import("/home/runner/work/code/code/node_modules/.pnpm/rollup@4.7.0/node_modules/rollup/dist/rollup").PluginContext' but required in type 'import("/home/runner/work/code/code/node_modules/.pnpm/rollup@3.29.4/node_modules/rollup/dist/rollup").PluginContext'.
   ELIFECYCLE  Command failed with exit code 2.
```

**Comment by vio** [comment]
@morriq thank you for creating the issue and for providing a failing example. I can replicate the issue as well.

**Comment by vio** [comment]
@morriq let me know if you get the chance to try `v0.2.3-beta.0` - https://stackblitz.com/edit/withastro-astro-mcgppu?file=astro.config.ts

**Comment by vio** [comment]
Fix released in `v0.2.3`, `rollup@4` was not included as a peer dependency and the package was bringing `rollup@3`

**Comment by morriq** [comment]
thank you @vio 🍺
